exporter/elasticsearchexporter/data_stream_router.go

```diff
@@ -53,15 +86,13 @@ func routeWithDefaults(defaultDSType string) func(
 			dataset = receiverName
 		}
 
-		// The naming convention for datastream is expected to be "logs-[dataset].otel-[namespace]".
-		// This is in order to match the built-in logs-*.otel-* index template.
-		if otel {
-			dataset += ".otel"
-		}
+		dataset = sanitizeDataStreamField(dataset, disallowedDatasetRunes, otel)
```

Comment:
set the otel dataset suffix here
```suggestion
		dataset = sanitizeDataStreamField(dataset, disallowedDatasetRunes, datasetSuffix)
```